crates/cargo-platform/tests/test_check_cfg.rs

```diff
@@ -0,0 +1,79 @@
+use cargo_platform::{CheckCfg, ExpectedValues};
+use std::collections::HashSet;
+
+#[test]
+fn check_cfg_none() {
+    let mut check_cfg = CheckCfg::default();
+
+    check_cfg.process_line("cfg_a");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_a").unwrap(),
+        ExpectedValues::Some(HashSet::from([None]))
+    );
+}
+
+#[test]
+fn check_cfg_empty() {
+    let mut check_cfg = CheckCfg::default();
+
+    check_cfg.process_line("cfg_b=");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_b").unwrap(),
+        ExpectedValues::Some(HashSet::from([]))
+    );
+}
+
+#[test]
+fn check_cfg_any() {
+    let mut check_cfg = CheckCfg::default();
+
+    check_cfg.process_line("cfg_c=any()");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_c").unwrap(),
+        ExpectedValues::Any
+    );
+
+    check_cfg.process_line("cfg_c");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_c").unwrap(),
+        ExpectedValues::Any
+    );
+}
+
+#[test]
+fn check_cfg_value() {
+    let mut check_cfg = CheckCfg::default();
+
+    check_cfg.process_line("cfg_d=\"test\"");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_d").unwrap(),
+        ExpectedValues::Some(HashSet::from([Some("test".to_string())]))
+    );
+
+    check_cfg.process_line("cfg_d=\"tmp\"");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_d").unwrap(),
+        ExpectedValues::Some(HashSet::from([
+            Some("test".to_string()),
+            Some("tmp".to_string())
+        ]))
+    );
+}
+
+#[test]
+fn check_cfg_weird() {
+    let mut check_cfg = CheckCfg::default();
+
+    check_cfg.process_line("cfg_e=\"double_quote\"\"");
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_e").unwrap(),
+        ExpectedValues::Some(HashSet::from([Some("double_quote\"".to_string())]))
+    );
+
+    check_cfg.process_line("cfg_e=any()"); // this shouldn't be possible but better handle this
+                                           // correctly anyway
+    assert_eq!(
+        *check_cfg.expecteds.get("cfg_e").unwrap(),
+        ExpectedValues::Any
+    );
+}
```

Comment:
Ideally, the weird cases would be named in the test name, calling out what behavior you are trying to show.

In this case, two unrelated behaviors are being tested, building on each other, which is harder to follow and brittle